File: USDA.java
import org.antlr.v4.runtime.atn.PredictionMode;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.ANTLRFileStream;
import java.io.IOException;
public class USDA {
	public static void main(String[] args) throws IOException {
	    System.err.println("Compiling file: "+args[0]);
	    CharStream input = new ANTLRFileStream(args[0]);
	    USDALexer lexer = new USDALexer(input);
	    CommonTokenStream tokens = new CommonTokenStream(lexer);
	    USDAParser parser = new USDAParser(tokens);
	    parser.getInterpreter().setPredictionMode(PredictionMode.SLL);
	    try {
		parser.usdFile();  // STAGE 1
	    } catch (Exception ex) {
		tokens.reset(); // rewind input stream
		parser.reset();
		parser.getInterpreter().setPredictionMode(PredictionMode.LL);
		parser.usdFile();  // STAGE 2
		// if we parse ok, it's LL not SLL
	    }
	}
}

File: USDA.mjs
import antlr4 from "antlr4"
import fs from "fs"
import USDALexer from "./USDALexer.js"
import USDAParser from "./USDAParser.js"
import USDAListener from "./USDAListener.js"
const iName = process.argv[2]
console.error("Parsing file", iName);
var input = fs.readFileSync(iName, 'UTF-8')
var chars = new antlr4.InputStream(input)
var lexer = new USDALexer(chars)
var tokens  = new antlr4.CommonTokenStream(lexer)
var parser = new USDAParser(tokens)
parser.buildParseTrees = true
var tree = parser.usdFile()

var extractor = new USDAListener()
antlr4.tree.ParseTreeWalker.DEFAULT.walk(extractor, tree)
